packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -203,6 +214,40 @@ export function authorizationServiceBuilder(
         )
       );
     },
+    async deleteClientKeyById({
+      clientId,
+      keyIdToRemove,
+      organizationId,
+      correlationId,
+      logger,
+    }: {
+      clientId: ClientId;
+      keyIdToRemove: string;
+      organizationId: TenantId;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<void> {
+      logger.info(`Removing key ${keyIdToRemove} from client ${clientId}`);
+
+      const client = await retrieveClient(clientId, readModelService);
+      assertOrganizationIsClientConsumer(organizationId, client.data);
+
+      retrieveKey(client.data, keyIdToRemove);
```

Comment:
```suggestion
      if (!client.data.keys.includes(keyIdToRemove)) {
        throw keyNotFound(keyIdToRemove, clientId);
      }
```
why not using only the check? Let's declare `retrieveKey` when we'll actually need it

Reply:
Sure, removed